byterun/major_gc.c

```diff
@@ -693,7 +694,12 @@ void caml_major_collection_slice (intnat howmuch)
   }
   if (p < dp) p = dp;
   if (p < caml_extra_heap_resources) p = caml_extra_heap_resources;
-  if (p > 0.3) p = 0.3;
+  p += p_backlog;
+  p_backlog = 0.0;
```

Comment:
Can you explain this change?  Is it strictly related to the new API, or could it impact the behavior of programs not using it?

Reply:
This is suggested in the last line of [MPR#7180](https://caml.inria.fr/mantis/view.php?id=7180) and repeated in [MPR#7750](https://caml.inria.fr/mantis/view.php?id=7750). If the extra_heap_resources ask for a huge slice (larger than one complete phase), we should probably not do just one phase, then forget about the rest.

I must confess I haven't tested the effects of this change in isolation.

To answer your second question, it does impact the behavior of programs that don't use the new API, hopefully in a positive way.

Reply:
Ok it makes sense, thanks for the explanation.